services/motion/builtin/move_attempt.go

```diff
@@ -108,8 +108,17 @@ func (ma *moveAttempt) start() error {
 // it cancels the processes spawned by it, drains all the channels that could have been written to and waits on processes to return.
 func (ma *moveAttempt) cancel() {
 	ma.cancelFn()
+
+	// Avoid race conditions by ensuring that the request is no longer executing before reading the channel.
+	// Position and obstacle threads are constantly waiting on a `ctx` so should not need this, but the request execution does not.
+	// As `FlushChan` will empty a channel with values or no-op an empty channel, if `ma.responseChan` is not yet written to then we will
+	// never return from `execute` and then `ma.backgroundWorkers.Wait()` wo;; hang.
```

Comment:
[sp] will* hang.  

Also general question are we supposed to be using backticks for code variables in comments?  No formatting is being applied

Reply:
Oops. What's a home row

The backticks for code variables in comments are a personal style choice. "are we supposed to" evaluates `false`.